### PR TITLE
Update go version to 1.20

### PR DESCRIPTION
**Pull request by tempoz**
Also update rules_go to 0.39.1.

Adjust flags handling in testing to future-proof against new flags being added to go's testing facilities.

**Related issues**: N/A

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3865** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/3865" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/3865?utm_source=stack-comment).

File: .github/workflows/checkstyle.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Setup go
         uses: actions/setup-go@v2
         with:
-          go-version: "1.19.3" # Keep in sync with WORKSPACE
+          go-version: "1.20.1" # Keep in sync with WORKSPACE
```
Comment (vadimberezniker) [inline .github/workflows/checkstyle.yaml:28]: There's been a newer version (1.20.3) released since this was created. Might as well update to the latest.

**Review by vadimberezniker — APPROVED** [review]
Looks good! Let's update to the latest minor release version.

File: .github/workflows/checkstyle.yaml (hunk 1 above)
Comment (tempoz) [reply .github/workflows/checkstyle.yaml]: Thanks for the catch; done.

**Pull request merged**